src/axiom_use.rs

```diff
@@ -0,0 +1,43 @@
+//! Generation of the `axiom_use` file.
+use crate::bit_set::Bitset;
+use crate::util::HashMap;
+use crate::StatementType;
+use crate::{as_str, database::time, Database};
+
+impl Database {
+    /// Writes a `axiom_use` file to the given writer.
+    pub fn write_axiom_use(&self, out: &mut impl std::io::Write) -> Result<(), std::io::Error> {
+        time(&self.options.clone(), "axiom_use", || {
+            let mut axiom_use_map = HashMap::default();
+            let mut axioms = vec![];
+            for sref in self.statements().filter(|stmt| stmt.is_assertion()) {
+                let label = sref.label();
+                let mut usage = Bitset::new();
+                if sref.statement_type() == StatementType::Provable {
+                    let mut i = 1;
+                    loop {
+                        let tk = sref.proof_slice_at(i);
+                        i += 1;
+                        if tk == b")" {
+                            break;
+                        }
+                        if let Some(usage2) = axiom_use_map.get(tk) {
+                            usage |= usage2
+                        }
+                    }
+                    write!(out, "{}:", as_str(label))?;
+                    for i in &usage {
+                        write!(out, " {}", axioms[i])?;
+                    }
+                    writeln!(out)?;
+                    axiom_use_map.insert(label, usage);
+                } else if label.starts_with(b"ax-") {
```

Comment:
Note that [`verify_markup`, line 82](https://github.com/tirix/metamath-knife/blob/a5a22c8bbefb20775982a915888062f0dcc4b45e/src/verify_markup.rs#L82), also has a reference to the `ax-` prefix. Ideally we would also include this in `$j` commands.